sklearn/preprocessing/_discretization.py

```diff
@@ -98,7 +98,7 @@ class KBinsDiscretizer(TransformerMixin, BaseEstimator):
         The edges of each bin. Contain arrays of varying shapes ``(n_bins_, )``
         Ignored features will have empty arrays.
 
-    n_bins_ : ndarray of shape (n_features,), dtype=np.int_
+    n_bins_ : ndarray of shape (n_features,), dtype=int
```

Comment:
I would rather use an explicit `np.int64`.

On my machine I have:

```python
>>> np.int_
<class 'numpy.int64'>
```

Reply:
I'm fine with int64.
But honestly, nobody will ever make use of 64-bit and int32 (or even better uint32) would be more than enough.